psi4/src/psi4/libmints/basisset.h

```diff
@@ -390,7 +396,7 @@ class BasisSet
      * @param py::dict      Python dictionary containing the basis information
      * @param forced_puream Force puream or not
     **/
-    static std::shared_ptr<BasisSet> construct_from_pydict(const std::shared_ptr <Molecule> &mol, py::dict pybs, const int forced_puream);
+    static std::shared_ptr<BasisSet> construct_from_pydict(const std::shared_ptr <Molecule> &mol, py::dict &pybs, const int forced_puream);
```

Comment:
Having said that, it'd probably be best to simply remove this function from BasisSet entirely, and just make it a static function that only lives in `export_mints.cc`.  That would avoid the janky forward declaration machinery in `basisset.h`.